### DISCUSSION TITLE
Possible migration wlsunset -> gammastep

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#87] [Porting for other distros]
@mac-a-r0ni @kuba3351 
I'm working on Hyprland support in nwg-panel, and I must say I like the compositor. Possibly in the future I may want all the shell to become Hyprland-compatible. As for now it seems that wlsunset doesn't work well here: I'm getting a couple of wlsunset processes that consume a lot of CPU power.

Would it be a problem to you if the shell started using [gammastep](https://gitlab.com/chinstrap/gammastep) instead one day?

**Comment by mac-a-r0ni** [comment]
Gammastep seems not on slackbuilds yet, so I'll look at it when I have some time, hopefully it's a easy package to get going! As for hyprland, I have not used it yet (seems I'll have to get that going as well) but I'm interested in it. So hopefully I can get them going this week and be ready for the option to exist soon.

**Reply by nwg-piotr** [reply]
Absolutely no hurry, and also no decisions have been taken yet on my side. The HyprlandTaskbar module is really a difficult piece of code, and just a first step out of ~~several~~ many.

**Reply by mac-a-r0ni** [reply]
I ended up going down a rabbit hole packaging hyprland for Slackware. It seems many deps are non-stable released software so far and it even uses a special waybar build (which to my eyes operates exactly the same as before) but it was pretty fun to play with for a bit. Nice animations and a good feel to hyprland. It would definitely be cool to see nwg-shell working with it some day. Doing all this I had forgotten all about gammastep in the process, so I'll need to get to that yet ;)

**Reply by nwg-piotr** [reply]
Current `master` branch of nwg-panel includes the HyprlandTaskbar module. Not yet tested enough, but seems to work pretty well. Also nwg-processes (an entry point of nwg-panel) have been modified to support Hyprland.

**Reply by mac-a-r0ni** [reply]
I will make some attempts to check it out!

**Comment by kuba3351** [comment]
For me, it is not a problem.

**Comment by nwg-piotr** [comment]
Thank you, guys. I'll let you know when/if it's necessary.

**Comment by rb-andrade** [comment]
Also created an APKBUILD to try this on alpine, did experience the increase in CPU power that mentioned from wlsunset. Although, hyprland itself seems to use more CPU poer than sway. Gammastep is packed for Alpine so I also see no problems in changing.

**Reply by nwg-piotr** [reply]
Well, there's no nwg-shell implementation on Alpine, as far as I know. Am I wrong?

> hyprland itself seems to use more CPU

I see no noticeable difference, but sway > 1.6 has always had bigger CPU usage on my hardware than previous versions.

**Reply by rb-andrade** [reply]
you are correct, nwg-shell is not currently packed for Alpine. I think only the components which use go language are packed and components using python are not packed. I have my own APKBUILD for nwg-shell.

**Comment by nwg-piotr** [comment]
Guys, nwg-shell-config 0.5.0 w/ Hyprland support is almost ready. I did split `main.py` into `main_sway.py` and `main_hyprland.py`. The Hyprland-related part uses gammastep, and I'm about to give up on wlsunset on sway as well. Starting from nwg-shell-config 0.5.0, the gammastep package will be required.

**Comment by mac-a-r0ni** [comment]
I’ll submit gammastep on my next updates (it’s been packaged and ready) but hyprland is stuck on .10 for Slackware 15.0, can run the latest on the dev version but it will be a month or more until the next stable release happens until I can ship hyprland support officially.   
  

  
I’ll get it up on my dev repo in the next few days. Looking forward to the official support!
  
  
  
— Jay Lanagan
  
  
  

  
  
>

**Comment by nwg-piotr** [comment]
Jay, I'm sorry about all the mess, but I think I'd like to stick to wlsunset. The most recent version seems to behave perfectly well on Hyprland, while gammastep keeps producing strange artifacts on my setup. Is wlsunset 0.3.0 available on Slackware yet?

**Comment by mac-a-r0ni** [comment]
It’s no problem to me. I maintain all the packages so the only limitation I have is what builds. As for wlsunset, Slackware 15.0 is stuck with 0.2.0 I believe because of the shipped wayland version. It’s not an issue on the dev release (which ships a newer wayland), as my repo is using the 0.3.0 version already. But for stable users, they are stuck on 0.2.0 (unless I can patch it to work on older wayland release).   
  

  
Keep in mind Slackware 15.0 also can’t use hyprland in its current state so any issues with it are moot because it’s not supported anyway.   
  

  
I haven’t sat down and caught up with the recent changes yet, in some hours that will be my target, but I anticipate no issues at all on Slackware-current. I need to work out some thing in hyprland packages I skipped over before I push changes out to anyone, but if you plan to stick with wlsunset, that’s already working as-is so I see no issues with it.   
  

  
  
  
— Jay Lanagan
  
  
  

  
  
>

**Comment by nwg-piotr** [comment]
Excellent. I'll update nwg-shell-config soon. It should work with no changes on sway, or at least I hope so.

**Comment by mac-a-r0ni** [comment]
Turns out wlsunset does build with the latest 0.3.0 release on 15.0. So along with the other updates (so far) I’ll push an update to that this week and all should be well.   
  

  
Unfortunately, the hyprland package isn’t in my control so I’ll see what can be done (and if the latest even builds) on 15. This is a package not easy to deal with since it relies on shipping a dev wlroots, so I’ll have to do some builds and see if I can’t work around that without messing up everything else that uses wlroots in the repo. Possibly adding another package to compensate. IIRC there’s a dep on qt6 for screensharing and I’m unsure on that entire can of worms at the moment but I’ll look into these things as I go.   
  
  
  
— Jay Lanagan
  
  
  

  
  
>
